DiscordBot/Backends/IBackend.cs

```diff
@@ -0,0 +1,14 @@
+using System.Threading.Tasks;
+
+namespace DiscordBot.Backends
+{
+    public interface IBackend<T>
+    {
+        public Task Run();
+
+        private Task MessageHandler(T e)

```

Comment:
Wait why is this function private, and why does it have a defualt implementation?
And why is it defined here in the interface in the first place???

Reply:
cause every backend has a message handlers, it has to have one to interact with new messages

and i dunno default implementation cause why not lmao i'll delete it

Reply:
and i don't see a reason it shouldn't be private but whatever i'll make it public - please explain why does that matter